src/ucs/stats/stats.c

```diff
@@ -477,95 +481,108 @@ void ucs_stats_node_free(ucs_stats_node_t *node)
-                (const char*)UCS_STATS_AGGRGT_SUM_CNT_NAMES_DB);
-
-        cnt_agrgt_sum_index = offset;
-        ucs_for_each_bit(counter_index, filter_node->counters_bitmask) {
-            ucs_list_for_each(temp_node, &filter_node->type_list_head,
-                              type_list) {
-                ucs_stats_context.aggrgt_cnt_names_db[cnt_agrgt_sum_index].name =
-                        node->cls->counter_names[counter_index];
-                ucs_stats_context.aggrgt_cnt_names_db[cnt_agrgt_sum_index]
-                        .class_name = node->cls->name;
-            }
-            cnt_agrgt_sum_index++;
+        capacity = ucs_array_capacity(&ucs_stats_context.aggrgt_sum_db);
+        ucs_array_set_length(&ucs_stats_context.aggrgt_sum_db, capacity);
+    }
+
+    element = &ucs_array_elem(&ucs_stats_context.aggrgt_sum_db, db_size);
+    printf("element = %p\n", element);
+
+    /* Initialize entry */
+    ucs_snprintf_safe(element->class_name, sizeof(element->class_name),
+                      "%s", class_name);
+    element->aggregate_sum_class_offset = ucs_stats_context.aggrgt_sum_db_last_offset;
+
+    /* Initialize return value */
+    offset = ucs_stats_context.aggrgt_sum_db_last_offset;
+
+    /* Return class_id */
+    node->cls->class_id = db_size;
+
+    /* Update global offset */
+    ucs_stats_context.aggrgt_sum_db_last_offset += n_cnts_in_cls;
+
+    /* Reallocate counters names db */
+    db_size = ucs_stats_context.aggrgt_cnt_names_db_size += n_cnts_in_cls;
+
+    total_alloc_size_bytes =
+            (db_size * sizeof(*ucs_stats_context.aggrgt_cnt_names_db));
+    ucs_stats_context.aggrgt_cnt_names_db = ucs_realloc(
+            ucs_stats_context.aggrgt_cnt_names_db, total_alloc_size_bytes,
+            (const char*)"stats_cnt_name_db");
+
+    cnt_index = offset;
+    ucs_for_each_bit(counter_index, filter_node->counters_bitmask) {
+        ucs_list_for_each(temp_node, &filter_node->type_list_head,
+                          type_list) {
+            ucs_stats_context.aggrgt_cnt_names_db[cnt_index].name =
+                    node->cls->counter_names[counter_index];
+            ucs_stats_context.aggrgt_cnt_names_db[cnt_index].class_name =
+                    node->cls->name;
         }
+        cnt_index++;
+    }
+
+    ucs_stats_context.aggrgt_sum_db_size++;
+
+    return offset;
+}
+
+static uint32_t
+ucs_stats_clsid_to_base_offset(ucs_stats_node_t *node,
+                                             size_t num_counters_in_class)
+{
+    uint32_t offset = 0;
+    ucs_stats_aggrgt_sum_clsid_to_name_db_t *element;
 
-        ucs_stats_context.aggrgt_sum_db_size++;
+    pthread_mutex_lock(&ucs_stats_context.lock);
+
+    if (ucs_likely(node->cls->class_id != 0)) {
+        element = &ucs_array_elem(&ucs_stats_context.aggrgt_sum_db,
+                                  node->cls->class_id);
+        offset = element->aggregate_sum_class_offset;
```

Comment:
align = on column

Reply:
Please explain the comment.

Reply:
code style - in consecutive lines = should be on same column
should be found in clang-format output

Reply:
I see, thanks! Fixed.